feat: add delete ssm parameter

johnnyrockets opened the pull request:
Add a new subcommand that allows for the deletion of ssm parameters by name.

```
parms delete
```

You will be prompted to select an existing parameter.  Select one that is safe to delete.

A confirmation prompt will be displayed.  Press y|Y to proceed.  Any other key will abort the deletion.

```
✔ Select parameter (type for fuzzy search): · delete-me-param
Delete parameter `delete-me-param`? [y/N]: y
Successfully deleted `delete-me-param`
```

johnnyrockets committed:
feat: add delete ssm parameter

Add a new subcommand that allows for the deletion of ssm parameters by name.

icevans commented:
Do you think delete should optionally work like fetch and edit, where you can fuzzy search for the parameter to delete? Otherwise, are we really offering much value over the AWS CLI's already available delete command which looks like this:

```
aws ssm delete-parameter --name "MyParameter"
```

johnnyrockets committed:
feat: make delete us fuzzy match

Rather than provide a name parameter to the delete subcommand, use the fuzzy
matching search to locate the parameter and then prompt the user to confirm
their selection before deleting.

johnnyrockets commented:
Updated to perform a fuzzy search and prompt the user for confirmation.

icevans reviewed (changes requested):
This is starting to look awesome! Instead of the manual and custom confirmation prompt, would you be open to using the confirmation prompt that comes included with the dialoguer crate that this project is using?

https://docs.rs/dialoguer/latest/dialoguer/struct.Confirm.html

johnnyrockets committed:
refactor: use dialoguer for confirm prompt

icevans reviewed (approved):
This works great, thanks for contributing!
